### PR TITLE
Rename legacy token for contribution id to match sched reminder tokens

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Rename legacy token for contribution id to match sched reminder tokens

Before
----------------------------------------
When choosing a token for contribution id from the widget {contribution.contribution_id} is inserted. This works from send letter but not from scheduled reminders

After
----------------------------------------
When choosing a token for contribution id from the widget {contribution.id} is inserted - this works from both places

Screenshots are the tokens and output from send letter (scheduled reminders is unchanged other than the addition of test cover by this PR)

image

image

Technical Details
----------------------------------------
In this case it turns out

1) the token being advertised by scheduled reminders is more correct ('id')
2) the legacy method resolves either 'id' or 'contribution_id' and
3) changing the advertised token from 'contribution.contribution_id' to contribution.id
does not cause contribution_id to stop working

So in summary - the advertised token was working in 'send letter' but not
in scheduled reminders. And changing which is advertised doesn't stop
the other from working in send letter

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
Just noting we have a letter unit test that should be covering the contribution_id token here https://github.com/civicrm/civicrm-core/blob/master/tests/phpunit/CRM/Contribute/Form/Task/PDFLetterCommonTest.php#L295

**Comment by seamuslee001** [comment]
This seems fine to me merging
